src/arrayMethodSort.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 1; j < this.length; j++) {
+        const previous = this[j - 1];
+        const next = this[j];
+
+        if ((!compareFunction && previous.toString() > next.toString())
+          || (compareFunction && compareFunction(previous, next) > 0)) {
+          this[j] = previous;
+          this[j - 1] = next;
+        }
+      }
+    }
```

Comment:
we shouldn't hardcode the iteration count in the external loop. think about a situation when array became sorted earlier